### PR TITLE
Denies apply and notifies the user for drafts and closed PRs.

### PR DESCRIPTION
**Pull request by ellisonmarks**

### CODE REVIEW

File: .github/workflows/tacos_apply.yml [hunk 1]
```diff
@@ -113,6 +113,13 @@ jobs:
             jq 'map(.author.login) | join(", ")' -r |
             gha-set-env 'GITHUB_PR_APPROVERS' \
           ;
+          gh pr view \
+              ${{ github.event.pull_request.html_url }} \
+              --json state \
+              --jq .state |
+            gha-set-env 'GITHUB_PR_STATE' \
+          ;
+
```
Comment (bukzor) [inline .github/workflows/tacos_apply.yml:122]: This has gotten non-trivial.
Let's get it into a real bash script.
Perhaps: `lib/ci/github-pr-env`

File: lib/ci/tacos-apply [hunk 2]
```diff
@@ -15,6 +15,32 @@ EOF
...
+    gha-step-summary >/dev/null <<EOF
+### TACOS Apply: $TF_ROOT_MODULE
+
+Applies are only allowed for open PRs.
+You may try reopening the PR first.
+
+Apply request refused: PR is not open
+EOF
+    exit 1
+fi
```
Comment (bukzor) [inline lib/ci/tacos-apply:42]: We've hit the "rule of three" here -- time to DRY out with a function:

```
refuse() {
    reason="$1"
   (
   echo "### TACOS Apply: $TF_ROOT_MODULE"
   cat  # pass deeper explanation via stdin
   echo
   echo "Apply request refused: $reason"
EOF
) | gha-step-summary >/dev/null
    exit 1
}

...

refuse "PR is not open" <<EOF
Cannot apply closed PRs.
Please re-open the PR if it has not been applied.
EOF
```

File: lib/ci/tacos-apply [hunk 3]
```diff
@@ -15,6 +15,32 @@ EOF
...
 
+if "$GITHUB_PR_DRAFT"; then
+    : no applying a draft! bail.
+    gha-step-summary >/dev/null <<EOF
+### TACOS Apply: $TF_ROOT_MODULE
+
+Applies are not allowed for draft PRs.
+Please set the PR ready for review first.
+
+Apply request refused: PR is a draft
```
Comment (bukzor) [inline lib/ci/tacos-apply:26]: I tried to touch up the wording. Feel free to just "ack".


```
Draft PRs cannot be applied.
Please try again after code review.

Apply request refused: PR is not ready
```

**Review by bukzor — APPROVED** [review]
approved: fixnship

File: lib/ci/tacos-apply (hunk 3 above)
Comment (ellisonmarks) [reply lib/ci/tacos-apply]: used this wording

File: lib/ci/tacos-apply (hunk 2 above)
Comment (ellisonmarks) [reply lib/ci/tacos-apply]: function added.

File: .github/workflows/tacos_apply.yml [hunk 4]
```diff
@@ -61,7 +61,6 @@ jobs:
     # note: we want to noop if they converted to draft first
     if: |
       true
-      && github.event.pull_request.draft == false
```
Comment (bukzor) [inline .github/workflows/tacos_apply.yml:64]: this seems like an error

**Review by bukzor — CHANGES_REQUESTED** [review]

**Review by bukzor — APPROVED** [review]

**Pull request merged**